cmd/client_changefeed.go

```diff
@@ -359,10 +355,12 @@ func verifyChangefeedParamers(ctx context.Context, cmd *cobra.Command, isCreate
 		CreatorVersion:    version.ReleaseVersion,
 	}
 
+	// user is not allowed to set sort-dir at changefeed level
 	if sortDir != "" {
-		cmd.Printf("[WARN] --sort-dir is deprecated in changefeed settings. " +
-			"Please use `cdc server --sort-dir` if possible. " +
-			"The sort-dir here will be no-op\n")
+		cmd.Printf(color.HiYellowString("[WARN] --sort-dir is deprecated in changefeed settings. " +
+			"Please use `cdc server --data-dir` to start the cdc server if possible, sort-dir will be set automatically. " +
+			"The --sort-dir here will be no-op\n"))
+		return nil, errors.New("Creating changefeed with `--sort-dir`, it's invalid")
```

Comment:
Is it expected that the user CANNOT create a changefeed with `sort-dir`? Is it not contradictory to the code in `setUpDataDir`?

Reply:
In `setUpDataDir`, we check changefeed set `sort-dir` or not, is for old changefeeds, which may already exist from the older version of the TiCDC. In this, we ban the `sort-dir` in changefeed level.